pkg/vtpm/src/server.cpp

```diff
@@ -403,28 +446,68 @@ handleRequest (int sock, google::protobuf::uint32 size)
     //fall through for cleanup and exit
 
 cleanup_and_exit:
-    //Clear all the files: inputFiles and expectedFiles
-    for (int i=0; i < request.expectedfiles_size(); i++) {
-        std::string expectedFile = request.expectedfiles(i);
-        unlink(expectedFile.c_str());
-    }
-    for (int i=0; i < request.inputfiles_size(); i++) {
-        const eve_tools::File& inputFile = request.inputfiles(i);
-        unlink(inputFile.name().c_str());
-    }
     if (cmdOut) {
         cmdOut.close();
     }
+    // Remove the working directory
+    rmClientWorkingDir(clientWorkingDir.c_str());
     return rc;
 }
 
+//Starting point of client threads
+static void
+serveClient(int sock, sockaddr_in clientAddr)
+{
+    ssize_t recvBytes = 0;
+    struct timeval tv;
+    char hdrBuf[CODED_STRM_HDR_LEN];
+    char *pBuf = hdrBuf;
+
+    if (sock < 0) {
+       return;
+    }
+
+    //Don't block forever on reading. Have a timeout
+    tv.tv_sec = CLIENT_READ_TIMEOUT;
+    tv.tv_usec = 0;
+    if (setsockopt(sock, SOL_SOCKET, SO_RCVTIMEO,
+              (const char*)&tv, sizeof tv) < 0) {
+        int rc = errno;
+        cerr << "Error setting recv timeout. client sock/err:"
+             << sock << strerror(rc) << std::endl;
+        return;
+    }
+
+    //create a client specific directory of the form:
+    //client_wd_x.x.x.x_yyyy, where x.x.x.x is source ip, and yyyy is source port.
```

Comment:
Why do you need the source port? This means that if the connection is dropped and re-restablished it will be treated as a new client which is a bit odd

Reply:
Here client refers to the TCP client and not the domU instance. Each vTPM API / command 
from domU creates a short-lived TCP connection to vTPM server. So we can have multiple TCP 
sessions active from a given domU at the same time. e.g. IoTEdge application and a shell program
 both invoking TPM requests at the same time. To avoid collision, source port is used as a qualifier in
 the working directory name.

Reply:
OK